Update Telephony Hook Guide (#4692)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/telephony-inline-hook/main/index.md

## Before you begin

* Review the information and required steps in the [Common Hook set-up steps](https://developer.okta.com/docs/guides/common-hook-set-up-steps/nodejs/main/) page first. This includes remixing (copying) the Glitch.com project used in this guide and understanding how Glitch projects are configured and used with hooks.

> **Note:** You can also use Amazon Web Services Lambda Serverless Framework as an external service. See [Setting Up Serverless Framework With AWS](https://www.serverless.com/framework/docs/getting-started).

> * Make sure you have an user in your org with a Phone authenticator enrolled. See [MFA Usage report](https://help.okta.com/okta_help.htm?type=oie&id=ext-mfa-usage).

Reviewer: ```suggestion
* Make sure you have a user in your org with a Phone authenticator enrolled. See [MFA Usage report](https://help.okta.com/okta_help.htm?type=oie&id=ext-mfa-usage).
```

* Make sure you have an active phone number in Twilio with SMS and MMS capabilities.

* Create a [TwiML bin](https://www.twilio.com/docs/runtime/tutorials/twiml-bins#create-a-new-twiml-bin) in your Twilio account for use with voice call messages. You'll use the automatically generated handler URL as a variable. Additionally, include an `otp` tag key within double brackets in the prepopulated XML. This tag key references the dynamic `otp` used later in this exercise. For example:

    ```xml
    <?xml version="1.0" encoding="UTF-8"?>
    <Response><Say>Your code is {{otp}}</Say></Response>
    ```

## About telephony inline hook implementation

> The Okta telephony inline hook allows you to integrate your own custom code into Okta flows that send SMS or voice call messages (except Okta Verify enrollment). You can integrate this hook with enrollment, authentication, and recovery flows that involve phone authenticators. Okta uses your external provider to deliver the one-time passcode (OTP) to the Requester. The provider can respond with commands that indicate if the delivery was successful or not.

Reviewer: ```suggestion
The Okta telephony inline hook allows you to integrate your own custom code into Okta flows that send SMS or voice call messages (except Okta Verify enrollment). You can integrate this hook with enrollment, authentication, and recovery flows that involve phone authenticators. Okta uses your external provider to deliver the one-time passcode (OTP) to the Requester. The provider can respond with commands that indicate if the delivery was successful.
```

## Response sent to the user

The following code is used to send the SMS or voice call to the user:

<StackSelector snippet="sendsmsmakecall" noSelector/>

## Send response to Okta

To tell Okta that the SMS or voice call message was successfully sent, return a `commands` object in the body of your HTTPS response. This object is an array that allows you to send multiple commands. The two required properties are `type` and `value`. The`value` property is where you specify the status of your telephony transaction and other relevant transaction metadata. The action type is `com.okta.telephony.action`.

<StackSelector snippet="responsetookta" noSelector/>

## Preview and test

> The external service example is now ready with code to receive and respond to an Okta call. The Okta org is now set up to call the external service using a telephony inline hook. In your Okta org, you can preview the JSON-formatted request and response directly from the Admin Console. You can also test the code directly in your org.

Reviewer: ```suggestion
The external service example is now ready with code to receive and respond to an Okta call. The Okta org is set up to call the external service using a telephony inline hook. In your Okta org, preview the JSON-formatted request and response directly from the Admin Console. You can also test the code directly in your org.
```

### Preview

To preview the telephony inline hook:

1. In the Admin Console, go to **Workflow** > **Inline Hooks**.

> 1. Select the telephony inline hook you created (in this example, **Twilio Telephony Hook**).

Reviewer: avoid past tense
```suggestion
1. Select the telephony inline hook you set up (in this example, **Twilio Telephony Hook**).
```

1. Select the **Preview** tab.
1. Define a value for `data.userProfile` by selecting a user in your org from the **data.userProfile** dropdown list.
1. Define a value for `requestType` by selecting a flow to test. In this example, select **MFA Verification**.

    > **Note**: If your user hasn't enrolled a phone authenticator, you can manually specify a phone number in the **Preview example inline hook request** section. Click **Edit** and add a value for the `phoneNumber` in the `messageProfile` section of the request (for example, `"+15555551212"`).

> 1. From the **Preview example inline hook request** section, click **Generate Request**. The request sent to the external service is displayed in JSON format.

Reviewer: displays requires a direct object 
```suggestion
1. From the **Preview example inline hook request** section, click **Generate Request**. The request sent to the external service appears in JSON format.
```

> 1. From the **View Service's Response** section, click **View Response**. A response from the external service is displayed in JSON format. Upon a successful response, an SMS code or voice call message with the code is sent to the associated user. Upon an error, the error message is returned in the response.

Reviewer: displays requires a direct object 
```suggestion
1. From the **View Service's Response** section, click **View Response**. A response from the external service appears in JSON format. Upon a successful response, an SMS code or voice call message with the code is sent to the associated user. Upon an error, the error message is returned in the response.
```
